### PR TITLE
camera snapshot

### PR DESCRIPTION
**Pull request by ToMadoRe**

### CODE REVIEW

File: cam_snapshot/package.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0"?>
+<package>
+  <name>cam_snapshot</name>
+  <version>0.0.0</version>
+  <description>The cam_snapshot package</description>
+
+  <!-- One maintainer tag required, multiple allowed, one person per tag --> 
+  <!-- Example:  -->
+  <!-- <maintainer email="jane.doe@example.com">Jane Doe</maintainer> -->
+  <maintainer email="strands@todo.todo">strands</maintainer>
```
Comment (marc-hanheide) [inline cam_snapshot/package.xml]: May I suggest that for packages that we potentially release (like in strands_utils) we fill these sections?

File: cam_snapshot/src/snapshot_node.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+{
+	ROS_INFO("I have received a new snapshot of the scene.");
+	PointInTPtr pScenePCl (new pcl::PointCloud<pcl::PointXYZRGB>());
+	pcl::fromROSMsg(msg, *pScenePCl);
+	std::stringstream filename;
+	filename << path_ << "/" << std::setw(5) << std::setfill('0') << pcl_in_curr_dir_ << ".pcd";
+	ROS_INFO("Saved pointcloud to %s", filename.str().c_str());
+	pcl::io::savePCDFileBinary(filename.str(), *pScenePCl );
+	pcl_in_curr_dir_++;
+	sub_pc = n_->subscribe("dummy_topic_to_unsuscribe", 1, &SnapshotNode::pclCallback, this); // is there a method to unsuscribe from a topic?
```
Comment (marc-hanheide) [inline cam_snapshot/src/snapshot_node.cpp]: can you explain the rationale of this line? If you ever want to unsubscribe, destroy you subscription handle. Quoting from the ROS docs: **`subscribe()` returns a Subscriber object that you must hold on to until you want to unsubscribe. When all copies of the Subscriber object go out of scope, this callback will automatically be unsubscribed from this topic.** I still don't quite get the purpose of this one...

**Comment by cdondrup** [comment]
Is this still a valid PR? It's quite old.

Found the following error:

```
/home/cdondrup/ros-ws/test/src/strands_utils/cam_snapshot/src/snapshot_node.cpp: In member function ‘void SnapshotNode::pclCallback(const PointCloud2&)’:
/home/cdondrup/ros-ws/test/src/strands_utils/cam_snapshot/src/snapshot_node.cpp:5:11: error: no match for ‘operator=’ in ‘((SnapshotNode*)this)->SnapshotNode::sub_pc = 0l’
/home/cdondrup/ros-ws/test/src/strands_utils/cam_snapshot/src/snapshot_node.cpp:5:11: note: candidate is:
/opt/ros/hydro/include/ros/subscriber.h:46:19: note: ros::Subscriber& ros::Subscriber::operator=(const ros::Subscriber&)
/opt/ros/hydro/include/ros/subscriber.h:46:19: note:   no known conversion for argument 1 from ‘long int’ to ‘const ros::Subscriber&’
```

ROS does not like setting the subscriber to NULL. Just calling `sub_pc.shutdown();` does the trick. @ToMadoRe Please fix and I'll merge or close if not relevant anymore.

**Comment by hawesie** [comment]
@ToMadoRe please respond to @cdondrup

**Comment by ToMadoRe** [comment]
As there are more advanced tools for this in the STRANDS repo already, I think it is not necessary to pull. For a quick and simple way to save some point clouds from the openni stream, it was quite handy though.

**Pull request closed without merging**